sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -691,10 +691,10 @@ public virtual Response UpsertEntity<T>(T entity, TableUpdateMode mode = TableUp
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
         /// <exception cref="RequestFailedException">The server returned an error. See <see cref="Exception.Message"/> for details returned from the server.</exception>
         /// <returns>The <see cref="Response"/> indicating the result of the operation.</returns>
-        public virtual async Task<Response> UpdateEntityAsync(IDictionary<string, object> entity, string ifMatch, TableUpdateMode mode = TableUpdateMode.Merge, CancellationToken cancellationToken = default)
+        public virtual async Task<Response> UpdateEntityAsync(IDictionary<string, object> entity, ETag ifMatch, TableUpdateMode mode = TableUpdateMode.Merge, CancellationToken cancellationToken = default)
```

Comment:
As a general question, Is `ETag` usually a separate parameter? Java had their `if-Match` as a boolean, so I think their method used the ETag inside the entity. Wondering what's the reasoning behind separating the ETag object as a parameter if it's a part of the entity.

(And wondering how metadata other than the ETag would be handled in entities.)